rosbag2/src/rosbag2/info.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rosbag2/info.hpp"
+
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rosbag2_storage/metadata_io.hpp"
+
+namespace rosbag2
+{
+
+Info::Info(std::shared_ptr<rosbag2_storage::Rosbag2StorageFactory> storage_factory)
```

Comment:
Shouldn't this be inverse? That the StorageFactory can be asked to get the info and not Info asking Storage?

Reply:
See https://github.com/ros2/rosbag2/pull/42#discussion_r224424391